src/main/java/org/prgrms/prgrmsspring/entity/user/Customer.java

```diff
@@ -0,0 +1,83 @@
+package org.prgrms.prgrmsspring.entity.user;
+
+import org.prgrms.prgrmsspring.exception.ExceptionMessage;
+
+import java.util.Objects;
+import java.util.UUID;
+import java.util.regex.Pattern;
+
+
+public class Customer {
+    private final UUID customerId;
+    private final String name;
+    private final String email;
+    private final Boolean isBlack;
+
+    private static final String NAME_REGEX = "^[a-zA-Z가-힣]*$"; // 영어 또는 한글만 허용하는 정규식
+    private static final String EMAIL_REGEX = "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$"; // 이메일 형식을 검증하는 정규식
+
+    public Customer(UUID customerId, String name, String email, Boolean isBlack) {
+        if (!Pattern.matches(NAME_REGEX, name)) {
+            throw new IllegalArgumentException(ExceptionMessage.INVALID_NAME_INPUT.getMessage());
+        }
+        if (!Pattern.matches(EMAIL_REGEX, email)) {
+            throw new IllegalArgumentException(ExceptionMessage.INVALID_EMAIL_INPUT.getMessage());
+        }
+        this.customerId = customerId;
+        this.name = name;
+        this.email = email;
+        this.isBlack = isBlack;
+    }
+
+    public Customer(UUID customerId, String name, String email) {
+        if (!Pattern.matches(NAME_REGEX, name)) {
+            throw new IllegalArgumentException(ExceptionMessage.INVALID_NAME_INPUT.getMessage());
+        }
+        if (!Pattern.matches(EMAIL_REGEX, email)) {
+            throw new IllegalArgumentException(ExceptionMessage.INVALID_EMAIL_INPUT.getMessage());
+        }
+        this.customerId = customerId;
+        this.name = name;
+        this.email = email;
+        this.isBlack = false;
+    }
```

Comment:
```suggestion
private static final Pattern NAME_PATTERN = Pattern.compile(NAME_REGEX);
    private static final Pattern EMAIL_PATTERN = Pattern.compile(EMAIL_REGEX);

    public Customer(UUID customerId, String name, String email) {
        this(customerId, name, email, false);
    }

    public Customer(UUID customerId, String name, String email, Boolean isBlack) {
        if (!NAME_PATTERN.matcher(name).matches()) {
            throw new IllegalArgumentException(ExceptionMessage.INVALID_NAME_INPUT.getMessage());
        }
        if (!EMAIL_PATTERN.matcher(email).matches()) {
            throw new IllegalArgumentException(ExceptionMessage.INVALID_EMAIL_INPUT.getMessage());
        }
        this.customerId = customerId;
        this.name = name;
        this.email = email;
        this.isBlack = isBlack;
    }
```

1. 정규식에 대한 검증을 할 때는 Pattern 클래스를 활용하게 되는데 이 클래스는 생성 비용이 비싸, 인스턴스 필드로 선언하여 재활용하는 것을 권장합니다.
2. 오버로딩 활용하시면 중복 코드 제거 및 초기화 시점을 명확하게 할 수 있을 것 같습니다.